service/dap: Add no-op handlers for unsupported requests

polinasok opened the pull request:
The client is notified that these requests are not supported by the false values for the corresponding capabilities in the 'initialize' response. No-op implementation is used because that is the default in vscode's codebase that defines the base classes for debug adaptors.
See https://github.com/microsoft/vscode-debugadapter-node/blob/master/adapter/src/debugSession.ts

derekparker committed:
pkg/proc: Add doc comments to Target Restart and Detach

polinasok committed:
Add no-op handlers for unsupported requests

eliben commented on service/dap/server_test.go:
```diff
@@ -216,6 +217,133 @@ func TestLaunchTestRequest(t *testing.T) {
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("\ngot  %#v\nwant %#v", got, want)
+		}
+		client.SetFunctionBreakpointsRequest()
```
Nit: add an empty line between each request test (after the closing `}` of the comparison `if`, to make them a little easier to distinguish visually

polinasok replied:
Done

polinasok committed:
Fix whitespace

aarzilli committed:
terminal: divide commands into categories

There are too many commands, for clarity they should be divided into
categories when printing and generating documentation.

polinasok committed:
logflags: Add pid to server-listening-at message (#1885)

* logflags: Add pid to server-listening-at message

* Address review comments: use logger

Co-authored-by: Polina Sokolova <polinasok@users.noreply.github.com>

aarzilli committed:
proc/native/linuc: Better native.Process.stop performance (#1874)

* proc/native/linux: only set breakpoints on threads that receive SIGTRAP

* proc/native/linux: do not call (*Thread).Stopped inside (*Process).stop

(*Thread).Stopped is slow because it needs to open, read and parse a
file in /proc, we don't actually need to do that, we can just rely on
the value of Thread.os.running.

Benchmark before:

BenchmarkConditionalBreakpoints-4              1        12476166303 ns/op

Benchmark after:

BenchmarkConditionalBreakpoints-4   	       1	10403533675 ns/op

Conditional breakpoint evaluation: 1.24ms -> 1ms

Updates #1549

polinasok commented:
Now tracked as https://github.com/go-delve/delve/pull/1918

polinasok committed:
Add couple of missing unsupported requests

chainhelen committed:
pkg/proc,service: support linux/386 (#1884)

Implement debugging function for 386 on linux with reference to AMD64.
There are a few remaining problems that need to be solved in another time.

1. The stacktrace of cgo are not exactly as expected.
2. Not implement `core` for now.
3. Not implement `call` for now. Can't not find `runtime·debugCallV1` or
   similar function in $GOROOT/src/runtime/asm_386.s.

Update #20

aarzilli committed:
proc,proc/*: add StopReason field to Target (#1877)

* proc,proc/*: move SelectedGoroutine to proc.Target, remove PostInitializationSetup

moves SelectedGoroutine, SwitchThread and SwitchGoroutine to
proc.Target, merges PostInitializationSetup with NewTarget.

* proc,proc/*: add StopReason field to Target

Adds a StopReason field to the Target object describing why the target
process is currently stopped. This will be useful for the DAP server
(which needs to report this reason in one of its requests) as well as
making pull request #1785 (reverse step) conformant to the new
architecture.

* proc: collect NewTarget arguments into a struct

polinasok committed:
service/dap: Add support for threads request (#1914)

* Add support for threads request

* Address review comments

* Relax threads test condition

* Address review comments

* Clean up unnecessary newline

* Respond to review comment

Co-authored-by: Polina Sokolova <polinasok@users.noreply.github.com>

aarzilli committed:
proc: cache result of GetG (#1921)

Benchmark before:

BenchmarkConditionalBreakpoints-4   	       1	7031242832 ns/op

Benchmark after:

BenchmarkConditionalBreakpoints-4   	       1	5282482841 ns/op

Conditional breakpoint evaluation latency: 0.70ms -> 0.52ms

Updates #1549

polinasok committed:
More comments

eliben commented:
Consider closing PRs here once they're tracked upstream. Any reason not to do that?

polinasok committed:
Sync with master
